net/golioth/settings.c

```diff
@@ -95,6 +96,27 @@ static int send_status_report(struct golioth_client *client,
 	return send_coap_response(client, response_buf, nbytes);
 }
 
+static void save_setting(const char *key, const struct golioth_settings_value *value)
+{
+	int err;
+	char name[64] = {};
+
+	/*
+	 * Use the "gsettings" package in the Zephyr settings subsystem
+	 * to store all settings.
+	 */
+	snprintf(name, sizeof(name), "gsettings/%s", key);
```

Comment:
`snprintf` doesn't guarantee to append `\0` at the end, if `key` will be large enough. I guess we should check return value from `snprintf` to make sure we fit into the `name` buffer.

From `man snprintf`:
> The  functions  snprintf() and vsnprintf() do not write more than size bytes (including the terminating null byte ('\0')).  If the output was truncated due to this limit, then the return value is the number of characters (excluding the terminating null byte) which would have been  written  to the  final  string  if enough space had been available.  Thus, a return value of size or more means that the output was truncated.

Reply:
Sure, I can pass `sizeof(name) - 1` to ensure the last byte is NULL.

I was using this as a reference, but I suppose the C++ standard deviates from C in this regard: https://cplusplus.com/reference/cstdio/snprintf/

> If the resulting string would be longer than n-1 characters, the remaining characters are discarded and not stored, but counted for the value returned by the function.

> A terminating null character is automatically appended after the content written.

Reply:
Still would be better to log an error/warning or even not attempt to write setting with truncated name.